src/dc_receive_imf.rs

```diff
@@ -685,11 +685,23 @@ async fn add_parts(
             }
         }
 
-        if !context.is_sentbox(server_folder).await?
+        // If the message is outgoing AND there is no Received header AND it's not in the sentbox,
+        // then ignore the email.
+        //
+        // We only apply this heuristic to classical emails, as it is not reliable (some servers
+        // such as systemli.org in June 2021 remove their own Received headers on incoming mails)
+        // and we know Delta Chat never stores drafts on IMAP servers.
+        let is_draft = !context.is_sentbox(server_folder).await?
             && mime_parser.get(HeaderDef::Received).is_none()
-        {
+            && mime_parser.get(HeaderDef::ChatVersion).is_none();
+
+        // Mozilla Thunderbird does not set \Draft flag on "Templates", but sets
+        // X-Mozilla-Draft-Info header, which can be used to detect both drafts and templates
+        // created by Thunderbird.
+        let is_draft = is_draft || mime_parser.get(HeaderDef::XMozillaDraftInfo).is_some();
```

Comment:
Maybe add a comment that this line does nothing currently but could be useful if we remove the Received: check later?

Reply:
Added a comment